src/webapi/window_or_worker.rs

```diff
@@ -29,4 +32,35 @@ pub trait IWindowOrWorker: ReferenceType {
             }, $3 );\
         ", self.as_ref().as_raw(), funcall_adapter::< F > as extern fn( *mut F ), callback, timeout );
     }
+
+    /// Sets a timer which executes a function once after the timer expires and can be cleared
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout)
+    // https://html.spec.whatwg.org/#windoworworkerglobalscope-mixin:dom-settimeout
+    fn set_clearable_timeout< F: FnOnce() + 'static >( &self, callback: F, timeout: u32 ) -> TimeoutHandle {
+        TimeoutHandle(js! (
+            const callback = @{Once(callback)};
+            return {
+                id: setTimeout(callback, @{timeout}),
+                callback
+            };
+        ).try_into().unwrap())
+    }
+}
+
+#[derive(Debug)]
+/// A reference to a timeout object created by set_clearable_timeout
+pub struct TimeoutHandle(Object);
+
+impl TimeoutHandle {
+    /// Clears a timer previously established by set_clearable_timeout
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/clearTimeout)
+    // https://html.spec.whatwg.org/#windoworworkerglobalscope-mixin:dom-clear-timeout
+     pub fn clear( & self ) {
+        js! { @(no_return)
+            clearTimeout(@{&self.0}.id);
+            @{&self.0}.callback.drop();
```

Comment:
It'll be more efficient to do something like this instead of serializing `self.0` twice (the `js!` macro is not smart enough to collapse duplicate interpolations):

```js
var val = @{&self.0};
clearTimeout(val.id);
val.callback.drop();
```